### PR TITLE
[4.x] Record entries after dd or exit

### PR DESCRIPTION
**Pull request by finagin**
PR for issue #1318

## Problem
When you use the `dd` function when the Telescope is enabled, you will not get the result because the entry entries happens in Laravel terminate. We don't see `dd` result and lose records of already completed database transactions, events, dumps, etc.

## Possible solution
- Always show dump result when `dd` called
- Record entries by self termination handler after `dd` or `exit` if config option `telescope.self_termination` is `true`

### CODE REVIEW

File: src/Watchers/DumpWatcher.php [hunk 1]
```diff
@@ -4,26 +4,26 @@
...
 
 class DumpWatcher extends Watcher
 {
     /**
      * The cache factory implementation.
-     *
-     * @var \Illuminate\Contracts\Cache\Factory
      */
-    protected $cache;
+    protected CacheFactory $cache;
```
Comment (driesvints) [inline src/Watchers/DumpWatcher.php:22]: Please revert all type and docblock changes you did.

**Review by driesvints — CHANGES_REQUESTED** [review]

File: src/Watchers/DumpWatcher.php (hunk 1 above)
Comment (finagin) [reply src/Watchers/DumpWatcher.php]: Done

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by finagin** [comment]
@taylorotwell, I’m feel absolutely certain that this code corrects a bug in the package. 

Now using the `dd` function looks like a failure: a white screen and no records in the Telescope. All because the recording occurs asynchronously in `$app->terminate`, but it does not reach it

**Pull request closed without merging**